Tests/Algorithm/Framework/Execution/StandardDeviationExecutionModelTests.cs

```diff
@@ -0,0 +1,151 @@
+        [TestCase(Language.Python, new[] { 250d, 250d, 250d }, 0, 0)]
+        public void OrdersAreSubmittedWhenRequiredForTargetsToExecute(
+            Language language,
+            double[] historicalPrices,
+            int expectedOrdersSubmitted,
+            decimal expectedTotalQuantity)
+        {
+            var actualOrdersSubmitted = new List<SubmitOrderRequest>();
+
+            var i = 1;
+            var time = new DateTime(2018, 8, 2, 16, 0, 0);
+            var historyProvider = new Mock<IHistoryProvider>();
+            historyProvider.Setup(m => m.GetHistory(It.IsAny<IEnumerable<HistoryRequest>>(), It.IsAny<DateTimeZone>()))
+                .Returns(historicalPrices.Select(x =>
+                    new Slice(time.AddMinutes(i++),
+                        new List<BaseData>
+                        {
+                            new TradeBar
+                            {
+                                Time = time.AddMinutes(i),
+                                Symbol = Symbols.AAPL,
+                                Open = Convert.ToDecimal(x),
+                                High = Convert.ToDecimal(x),
+                                Low = Convert.ToDecimal(x),
+                                Close = Convert.ToDecimal(x),
+                                Volume = 100m
+                            }
+                        })));
+
+            var algorithm = new QCAlgorithmFramework();
+            algorithm.SetPandasConverter();
+            algorithm.SetHistoryProvider(historyProvider.Object);
+            algorithm.SetDateTime(time.AddMinutes(5));
+
+            var security = algorithm.AddEquity(Symbols.AAPL.Value);
+            security.SetMarketPrice(new TradeBar { Value = 250 });
+
+            algorithm.SetFinishedWarmingUp();
+
+            var orderProcessor = new Mock<IOrderProcessor>();
+            orderProcessor.Setup(m => m.Process(It.IsAny<SubmitOrderRequest>()))
+                .Returns((SubmitOrderRequest request) => new OrderTicket(algorithm.Transactions, request))
+                .Callback((SubmitOrderRequest request) => actualOrdersSubmitted.Add(request));
+            orderProcessor.Setup(m => m.GetOpenOrders(It.IsAny<Func<Order, bool>>()))
+                .Returns(new List<Order>());
+            algorithm.Transactions.SetOrderProcessor(orderProcessor.Object);
+
+            var model = GetExecutionModel(language);
+            algorithm.SetExecution(model);
+
+            var changes = new SecurityChanges(new[] { security }, Enumerable.Empty<Security>());
+            model.OnSecuritiesChanged(algorithm, changes);
+
+            var targets = new IPortfolioTarget[] { new PortfolioTarget(Symbols.AAPL, 10) };
+            model.Execute(algorithm, targets);
+
+            Assert.AreEqual(expectedOrdersSubmitted, actualOrdersSubmitted.Count);
+            Assert.AreEqual(expectedTotalQuantity, actualOrdersSubmitted.Sum(x => x.Quantity));
```

Comment:
Do we want to assert on the _time_ the orders are placed? Or perhaps the individual order sizes?